docs/src/app/(public)/(content)/react/utils/use-render/page.mdx

```diff
@@ -0,0 +1,99 @@
+<Meta name="description" content="Utility for supporting a render prop in custom components." />
+
+The `useRender` hook allows you to support the `render` prop to override the default rendered element (similar to `asChild` from Radix) in your custom components.
+
+## API reference
+
+### Input parameters
+
+<PropsReferenceTable
+  data={{
+    render: {
+      type: 'RenderProp<State>',
+      description:
+        'The React element or a function that returns one to override the default element.',
+    },
+    props: {
+      type: 'Record<string, unknown>',
+      description:
+        'Props to be spread on the rendered element.\n\nThey are merged with the internal props\n of the component, so that event handlers are merged,\nclass names and styles are joined, and other external props overwrite the internal ones.',
+    },
+    state: {
+      type: 'State',
+      description:
+        'The state of the component. It will be used as a parameter for the render callback.',
+    },
+  }}
+/>
+
+### Return value
+
+The hook returns a function that when called returns the element that should be rendered.
+
+## Examples
+
+This is an example of a Text component supporting the `render` prop.
+
+<Demo path="./demos/render" />
+
+Developers can also use the callback version of the `render` prop to access the internal `state` of a component.
+
+<Demo path="./demos/render-callback" />
+
+### Merging the props
+
+You can use the `mergeProp` utility to safely merge the props.
+It takes care of concatenating the class names, as well as merging the style prop and the event handlers.
+It merges objects from left to right. Subsequent sources overwrite property assignments of previous sources.
+
+You can use it both in the `useRender` hook and the callback version of the render prop.
+
+```jsx {7-7} title="Using mergeProps in the render callback"
+import { mergeProps } from '@base-ui-components/react/merge-props';
+import styles from './index.module.css';
+
+function Button() {
+  return (
+    <Component
+      render={(props, state) => <button {...mergeProps(props, { className: styles.Button })} />}
+    />
+  );
+}
+```
+
+### Migrating from Radix
+
+Radix and Base UI take different approaches to customizing the rendered element within a component. Radix uses the `asChild` prop, while Base UI uses the `render` prop. Learn more about how composition works in Base UI in the [composition guide](/react/handbook/composition).
+
+In Radix, you'd use the `Slot` component to implement your own `asChild` prop.
+
+```jsx title="Custom asChild prop with Radix"
+import { Slot } from 'radix-ui';
+
+function Button({ asChild, ...props }) {
+  const Comp = asChild ? Slot.Root : 'button';
+  return <Comp {...props} />;
+}
+
+// Usage
+<Button asChild>
+  <a href="/contact">Contact</a>
+</Button>;
+```
+
+In Base UI, you use `useRender` to implement your own `render` prop. The example below is the equivalent implementation to the Radix example above.
```

Comment:
re "The example below is the equivalent implementation to the Radix example above", do you think this example should use the `mergeProps` util to be fully equivalent to the Radix example?

Reply:
Hmm, the example is to simple to show the usage of mergeProps as we are not defining any props in the component itself. Maybe the other examples should go in more details and we can keep this one simple, what do you think?

Reply:
👍